sdk/core/Azure.Core/src/Shared/PageResponseEnumerator.cs

```diff
@@ -11,40 +11,37 @@ namespace Azure.Core
 {
     internal static class PageResponseEnumerator
     {
-        public static IEnumerable<Response<T>> CreateEnumerable<T>(Func<string?, Page<T>> pageFunc) where T : notnull
+
+        public static FuncPageable<T> CreateEnumerable<T>(Func<string?, Page<T>> pageFunc) where T : notnull
         {
-            string? nextLink = null;
-            do
-            {
-                Page<T> pageResponse = pageFunc(nextLink);
-                foreach (T setting in pageResponse.Values)
-                {
-                    yield return Response.FromValue(pageResponse.GetRawResponse(), setting);
-                }
-                nextLink = pageResponse.ContinuationToken;
-            } while (nextLink != null);
+            return new FuncPageable<T>((continuationToken, pageSizeHint) => pageFunc(continuationToken));
+        }
+
+        public static FuncPageable<T> CreateEnumerable<T>(Func<string?, int?, Page<T>> pageFunc) where T : notnull
+        {
+            return new FuncPageable<T>(pageFunc);
```

Comment:
Do these need to allow threading a cancellationToken through?

Reply:
They would usually capture it as part of lambda.